oop/src/views/ReviewPage.vue

```diff
@@ -44,31 +50,58 @@ export default {
   },
   data: () => {
     return {
-      isLoading: false
+      isLoading: false,
+      isSystemUnavailable: false,
     }
   },
   methods: {
     submitForm() {
       this.isLoading = true;
-      
+      this.isSystemUnavailable = false;
+
+      this.$store.dispatch(formModule + '/' + SET_SUBMISSION_DATE, new Date());
+
       const token = this.$store.state.form.captchaToken;
+      const applicationUuid = this.$store.state.form.applicationUuid;
+      const formState = this.$store.state.form;
 
-      apiService.submitApplication(token)
-        .then(() => {
-          // Handle success.
-          logService.logSubmission({ message: 'Success', error: null }, 'TEMP_UUID', 'TEMP_REF_NUMBER');
-        })
-        .catch(() => {
-          // Handle error.
-          logService.logSubmission({ message: 'Error sending application', error: 'TEMP_ERROR' }, 'TEMP_UUID', 'TEMP_REF_NUMBER');
+      apiService.submitApplication(token, formState)
+        .then((response) => {
+          // Handle HTTP success.
+          const returnCode = response.data.returnCode;
+          const referenceNumber = response.data.referenceNumber;
+          const errorMessage = response.data.message;
+
+          this.isLoading = false;
+
+          switch (returnCode) {
+            case '0': // Submission successful.
+              logService.logSubmission({ message: 'Success', error: null }, applicationUuid, referenceNumber);
+              this.$store.dispatch(formModule + '/' + SET_REFERENCE_NUMBER, referenceNumber);
+              this.navigateToSubmissionPage();
+              break;
+            case '1': // Submission failed.
+              logService.logSubmission({ message: 'Submission failure', error: errorMessage }, applicationUuid, 'N/A');
+              this.navigateToSubmissionErrorPage();
+              break;
+            case '3': // System unavailable.
+              this.isSystemUnavailable = true;
+              logService.logSubmission({ message: 'Submission failure', error: errorMessage }, applicationUuid, 'N/A');
+              scrollToError();
+              break;
+          }
```

Comment:
Hi Harry, do we need to catch if the returnCode is 2 from the middleware?

Reply:
It wasn't listed in the confluence doc, but I've added it in, and handling it as an error (since it's not '0').